### PR TITLE
graph exporter and inference code

### PR DESCRIPTION
**Pull request by ziyi-yang**
@caihsiao @xiangxin-zhu

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# To update TensorFlow to a new revision.
+# 1. Update the 'git_commit' args below to include the new git hash.
+# 2. Get the sha256 hash of the archive with a command such as...
+#    curl -L https://github.com/tensorflow/tensorflow/archive/<git hash>.tar.gz | sha256sum
+#    and update the 'sha256' arg with the result.
+# 3. Request the new archive to be mirrored on mirror.bazel.build for more
+#    reliable downloads.
+load("//build_rules:repo.bzl", "tensorflow_http_archive")
+
+tensorflow_http_archive(
```
Comment (xiangxin-zhu) [inline WORKSPACE:10]: Please add a comment that the tensorflow version here must match the swift repo.

File: WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  name = "io_bazel_rules_closure",
+  sha256 = "a38539c5b5c358548e75b44141b4ab637bba7c4dc02b46b1f62a96d6433f56ae",
+  strip_prefix = "rules_closure-dbb96841cc0a5fb2664c37822803b06dab20c7d1",
+  urls = [
+    "https://mirror.bazel.build/github.com/bazelbuild/rules_closure/archive/dbb96841cc0a5fb2664c37822803b06dab20c7d1.tar.gz",
+    "https://github.com/bazelbuild/rules_closure/archive/dbb96841cc0a5fb2664c37822803b06dab20c7d1.tar.gz",  # 2018-04-13
+  ],
+)
+
+# Abseil
```
Comment (xiangxin-zhu) [inline WORKSPACE:30]: Delete Abseil.

File: WORKSPACE [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  name = "org_tensorflow",
+  sha256 = "cbd96914936ce3aacc39e02c2efb711f937f8ebcda888c349eab075185d7c914",
+  git_commit = "d8fac4cb80eb0c42d2550bcb720a80d29fc5f22d",
+  #sha256 = "783341cb4190db39166cd3ffb3b1fc590f93b7c5f95539819776f52cfebcb1ff",
+  #git_commit = "d752244fbaad5e4268243355046d30990f59418f",
+)
+
+# TensorFlow depends on "io_bazel_rules_closure" so we need this here.
+# Needs to be kept in sync with the same target in TensorFlow's WORKSPACE file.
+http_archive(
```
Comment (xiangxin-zhu) [inline WORKSPACE:20]: Same here, please comment that the version must match that in swift.

File: research/object_detection/inference_on_image.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+print("Successfully loaded the label map.")
+
+def load_image_into_numpy_array(image):
+    (im_width, im_height) = image.size
+    print("Image size: %d x %d" % (im_width, im_height))
+    return np.array(image.getdata()).reshape((im_height, im_width, 3)).astype(np.uint8)
+
+PATH_TO_TEST_IMAGES_DIR = 'test_images'
+TEST_IMAGE_PATHS = [ os.path.join(PATH_TO_TEST_IMAGES_DIR, 'image{}.jpg'.format(i)) for i in range(1, 3) ]
+PATH_TO_TEST_IMAGES_DIR = '/home/xiangxin/Downloads/test_image/test'
```
Comment (xiangxin-zhu) [inline research/object_detection/inference_on_image.py]: remove line 67.

File: research/object_detection/inference_on_image.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+# List of the strings that is used to add correct label for each box.
+PATH_TO_LABELS = "/home/deepdot/Dataset/Budweiser/label_map.pbtxt"
+#PATH_TO_LABELS = "src/third_party/tensorflow/models/research/object_detection/data/mscoco_label_map.pbtxt"
+
+NUM_CLASSES = 30
+
+
+detection_graph = tf.Graph()
+#with detection_graph.as_default():
```
Comment (xiangxin-zhu) [inline research/object_detection/inference_on_image.py]: remove line 46-50?

File: research/object_detection/inference_on_image.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+from object_detection.utils import ops as utils_ops
+
+if tf.__version__ < '1.4.0':
+    raise ImportError('Please upgrade your tensorflow installation to v1.4.* or later!')
+
+from utils import label_map_util
+from utils import visualization_utils as vis_util
+
+# What model to run.
+#MODEL_NAME = 'ssd_mobilenet_v1_coco_2017_11_17'
```
Comment (xiangxin-zhu) [inline research/object_detection/inference_on_image.py]: No MODEL_NAME is specified?

**Review by xiangxin-zhu — CHANGES_REQUESTED** [review]

File: WORKSPACE (hunk 1 above)
Comment (ziyi-yang) [reply WORKSPACE:10]: added.

File: WORKSPACE (hunk 3 above)
Comment (ziyi-yang) [reply WORKSPACE:20]: added.

File: WORKSPACE (hunk 2 above)
Comment (ziyi-yang) [reply WORKSPACE:30]: removed.

File: research/object_detection/inference_on_image.py (hunk 4 above)
Comment (ziyi-yang) [reply research/object_detection/inference_on_image.py]: removed.

File: research/object_detection/inference_on_image.py (hunk 5 above)
Comment (ziyi-yang) [reply research/object_detection/inference_on_image.py]: removed.

File: research/object_detection/inference_on_image.py (hunk 6 above)
Comment (ziyi-yang) [reply research/object_detection/inference_on_image.py]: User-specified path.

**Review by ziyi-yang — COMMENTED** [review]
Fixed path and cleaned.

**Review by xiangxin-zhu — APPROVED** [review]

**Pull request merged**